code/game/machinery/doors/firedoor.dm

```diff
@@ -259,18 +259,27 @@
 	if(!istype(C, /obj/item/forensics))
 		add_fingerprint(user)
 	if(operating)
-		return//Already doing something.
+		return // Already doing something.
 	if(C.iswelder() && !repairing)
-		var/obj/item/weldingtool/W = C
-		if(W.remove_fuel(0, user))
+		var/obj/item/weldingtool/WT = C
+		if(WT.isOn())
+			user.visible_message(
+				SPAN_WARNING("[user] begins welding [src] [blocked ? "open" : "shut"]."),
+				SPAN_NOTICE("You begin welding [src] [blocked ? "open" : "shut"]."),
+				SPAN_ITALIC("You hear a welding torch on metal.")
+			)
+			playsound(src, 'sound/items/welder.ogg', 50, 1)
+			if (!do_after(user, 2/C.toolspeed SECONDS, act_target = src, extra_checks = CALLBACK(src, .proc/is_open, src.density)))
+				return
+			if(!WT.remove_fuel(0,user))
```

Comment:
If it takes time it should also require some fuel to do that.

Reply:
fixed.

Reply:
just playtested to be sure. the code as-is already removes 1 unit of welding fuel when welding the emergency shutter shut or open.